### PR TITLE
Redmine 11387: Batch, Fix batch null unknown when no code entered in …

### PR DESCRIPTION
**Pull request by man-axelor**
…MailBatch, remove BaseBatch and its entry menu, because it is not used and the batch called an undefined method in BaseBatchController, and what was wanted in BaseBatch is in CRM batch

### CODE REVIEW

File: axelor-base/src/main/java/com/axelor/apps/base/web/MailBatchController.java [hunk 1]
```diff
@@ -41,16 +42,19 @@
...
 
 	public void remindTimesheet(ActionRequest request, ActionResponse response) throws AxelorException{
+	    try{
+	        MailBatch mailBatch = request.getContext().asType(MailBatch.class);
 
-		MailBatch mailBatch = request.getContext().asType(MailBatch.class);
+	        Batch batch = null;
 
-		Batch batch = null;
+	        batch = mailBatchService.remindMail(mailBatchRepo.find(mailBatch.getId()));
```
Comment (gdu-axelor) [inline axelor-base/src/main/java/com/axelor/apps/base/web/MailBatchController.java]: Why use two lines for assign batch ?

File: axelor-base/src/main/resources/domains/BaseBatch.xml [hunk 2]
```diff
@@ -1,21 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<domain-models xmlns="http://axelor.com/xml/ns/domain-models"
-  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://axelor.com/xml/ns/domain-models http://axelor.com/xml/ns/domain-models/domain-models_5.0.xsd">
-
-  <module name="base" package="com.axelor.apps.base.db"/>
-
-  <entity name="BaseBatch" lang="java">
```
Comment (gdu-axelor) [inline axelor-base/src/main/resources/domains/BaseBatch.xml]: Why remove this object ?

File: axelor-base/src/main/resources/domains/MailBatch.xml [hunk 3]
```diff
@@ -8,7 +8,7 @@
   <entity name="MailBatch" lang="java">
   	
   	<!-- HEADER -->
-  	<string name="code" title="Code" namecolumn="true" unique="true"/>
+    <string name="code" title="Code" namecolumn="true" unique="true" required="true"/>
```
Comment (gdu-axelor) [inline axelor-base/src/main/resources/domains/MailBatch.xml]: Code should be able to be null if the batch is run from UI. Only required if the batch is called per WS from a scheduler.

File: axelor-base/src/main/resources/domains/BaseBatch.xml (hunk 2 above)
Comment (man-axelor) [reply axelor-base/src/main/resources/domains/BaseBatch.xml]: Changes were made, now we don't remove the object, the undefined method actionTarget is now defined, code is not required.

**Pull request closed without merging**